task2/list.c

```diff
@@ -0,0 +1,151 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "list.h"
+
+typedef struct Node
+{
+    Value value;
+    struct Node *next;
+} Node;
+
+typedef struct List
+{
+    Node *head;
+} List;
+
+Error listCreate(List **list)
+{
+    *list = calloc(1, sizeof(List));
+    if (*list == NULL)
+    {
+        return MemoryAllocationError;
+    }
+    return OK;
+}
+
+Error addValue(List *list, Value value)
+{
+    if (list == NULL)
+    {
+        return PointerIsNull;
+    }
+    if (list->head == NULL)
+    {
+        list->head = calloc(1, sizeof(Node));
+        list->head->value = value;
+        return OK;
+    }
+
+    Node *newNode = calloc(1, sizeof(Node));
+    if (newNode == NULL)
+    {
+        return PointerIsNull;
+    }
+    newNode->value = value;
+
+    Node *currentNode = list->head;
+    Node *previousNode = NULL;
+    while (currentNode->value < newNode->value && currentNode->next != NULL)
```

Comment:
```suggestion
    while (currentNode!= NULL && currentNode->value < newNode->value)
```
При таком условии цикла рассмотренные ниже случаи можно объединить в один.
Причем порядок конъюнктов важен. Если `currentNode` равняется `NULL`, то второй конъюнкт не будет вычислен и ошибка не случится.